src/math/quaternion.hpp

```diff
@@ -35,4 +35,11 @@ D angleBetweenQuaternions(const Eigen::Quaternion< D> & q1,
   return theta;
 }
 
+template <typename D>
```

Comment:
I think it is not a good idea to put this function in such a file. You can let it in unit test.